src/schema/v2/artist/updateArtistMutation.ts

```diff
@@ -0,0 +1,67 @@
+import { GraphQLNonNull, GraphQLString } from "graphql"
+import { mutationWithClientMutationId } from "graphql-relay"
+import { ResolverContext } from "types/graphql"
+import { snakeCase } from "lodash"
+
+interface Input {
+  displayName: string
+  first: string
+  id: string
+  last: string
+  middle: string
+}
+
+interface GravityInput {
+  display_name: string
+  first: string
+  id: string
+  last: string
+  middle: string
+}
+
+interface GravityError {
+  statusCode: number
+  body: { error?: string; text?: string; message?: string }
+}
+
+export const updateArtistMutation = mutationWithClientMutationId<
+  Input,
+  any | null,
+  ResolverContext
+>({
+  name: "UpdateArtistMutation",
+  description: "Update the artist",
+  inputFields: {
+    displayName: { type: GraphQLString },
+    first: { type: GraphQLString },
+    id: { type: new GraphQLNonNull(GraphQLString) },
+    last: { type: GraphQLString },
+    middle: { type: GraphQLString },
+  },
+  outputFields: {},
+  mutateAndGetPayload: async (args, { updateArtistLoader }) => {
+    if (!updateArtistLoader) {
+      throw new Error(
+        "You need to pass a X-Access-Token header to perform this action"
+      )
+    }
+
+    const updateArtistLoaderPayload = Object.keys(args)
+      .filter((key) => key !== "id")
+      .reduce(
+        (acc, key) => ({ ...acc, [snakeCase(key)]: args[key] }),
+        {} as GravityInput
+      )
+
+    try {
+      return await updateArtistLoader?.(args.id, updateArtistLoaderPayload)
+    } catch (err) {
+      if ("body" in (err as any)) {
+        const e = err as GravityError
+        throw new Error(e.body.text ?? e.body.error ?? e.body.message)
+      }
+
+      throw err
+    }
+  },
+})
```

Comment:
Unfortunately @jonallured we'll need to update our mutations to support the `thingOrError` pattern. @mzikherman has been laying down the hammer on this one. Throw it into chat gpt if you don't feel like refactoring it by hand.

Reply:
😹 ⚒️

Reply:
Ok I think I did it! I opened https://github.com/artsy/metaphysics/blob/main/src/schema/v2/me/updateCollectionMutation.ts for reference and then renamed things as needed - does that look right?